mods/ra/maps/allies-08a/allies08a.lua

```diff
@@ -0,0 +1,132 @@
+--[[
+   Copyright 2007-2018 The OpenRA Developers (see AUTHORS)
+   This file is part of OpenRA, which is free software. It is made
+   available to you under the terms of the GNU General Public License
+   as published by the Free Software Foundation, either version 3 of
+   the License, or (at your option) any later version. For more
+   information, see COPYING.
+]]
+AlliedBoatReinforcements = { "dd", "dd" }
+TimerTicks = DateTime.Minutes(21)
+ObjectiveBuildings = { Chronosphere, AlliedTechCenter }
+
+InitialAlliedReinforcements = function()
+	Trigger.AfterDelay(DateTime.Seconds(2), function()
+		Media.PlaySpeechNotification(greece, "ReinforcementsArrived")
+		Reinforcements.Reinforce(greece, { "mcv" }, { MCVEntry.Location, MCVStop.Location })
+		Reinforcements.Reinforce(greece, AlliedBoatReinforcements, { DDEntry.Location, DDStop.Location })
+	end)
+end
+
+FinishTimer = function()
+	for i = 0, 5, 1 do
+		local c = TimerColor
+		if i % 2 == 0 then
+			c = HSLColor.White
+		end
+
+		Trigger.AfterDelay(DateTime.Seconds(i), function() UserInterface.SetMissionText("The experiment is a success!", c) end)
+	end
+	Trigger.AfterDelay(DateTime.Seconds(6), function() UserInterface.SetMissionText("") end)
+end
+
+DefendChronosphereCompleted = function()
+	local cells = Utils.ExpandFootprint({ ChronoshiftLocation.Location }, false)
+	local units = { }
+	for i = 1, #cells do
+		local unit = Actor.Create("2tnk", true, { Owner = greece, Facing = 0 })
+		units[unit] = cells[i]
+	end
+	Chronosphere.Chronoshift(units)
+
+	Trigger.AfterDelay(DateTime.Seconds(3), function()
+		greece.MarkCompletedObjective(DefendChronosphere)
+		greece.MarkCompletedObjective(KeepBasePowered)
+	end)
+end
+
+PowerFailure = function()
+	greece.MarkFailedObjective(KeepBasePowered)
+end
+
+MissionFailed = function()
+	greece.MarkFailedObjective(DefendChronosphere)
+end
+
+ticked = TimerTicks
+Tick = function()
+	ussr.Cash = 5000
+
+	if ussr.HasNoRequiredUnits() then
+		greece.MarkCompletedObjective(DefendChronosphere)
+		greece.MarkCompletedObjective(KeepBasePowered)
+	end
+
+	if greece.HasNoRequiredUnits() then
+		ussr.MarkCompletedObjective(BeatAllies)
+	end
+
+	if ticked > 0 then
+		UserInterface.SetMissionText("Chronosphere experiment completes in " .. Utils.FormatTime(ticked), TimerColor)
+		ticked = ticked - 1
+	elseif ticked == 0 and (greece.PowerState ~= "Normal") then
+		PowerFailure()
```

Comment:
Can you call `greece.MarkFailedObjective(KeepBasePowered)` directly here, and remove the one-line function, please?